server/users.go

```diff
@@ -1,317 +1,153 @@
-	users, err := store.All(ctx)
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
+	if r.Provider == "" {
+		return errors.New("Provider required on Chronograf User request body")
 	}
-
-	_, hasRoles := h.hasRoles(ctx, ts)
-	ur := make([]userResponse, len(users))
-	for i, u := range users {
-		usr := newUserResponse(srcID, u.Name).WithPermissions(u.Permissions)
-		if hasRoles {
-			usr.WithRoles(srcID, u.Roles)
-		}
-		ur[i] = *usr
+	if r.Scheme == "" {
+		return errors.New("Scheme required on Chronograf User request body")
 	}
+	return nil
+}
 
-	res := usersResponse{
-		Users: ur,
+func (r *userRequest) ValidUpdate() error {
+	if r.Name == "" && r.Provider == "" && r.Scheme == "" {
+		return errors.New("No fields to update")
 	}
-
-	encodeJSON(w, http.StatusOK, res, h.Logger)
+	return nil
 }
 
-// SourceUserID retrieves a user with ID from store.
-func (h *Service) SourceUserID(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	uid := httprouter.GetParamFromContext(ctx, "uid")
-
-	srcID, ts, err := h.sourcesSeries(ctx, w, r)
-	if err != nil {
-		return
-	}
-	store := ts.Users(ctx)
-	u, err := store.Get(ctx, uid)
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
-	}
+type userResponse struct {
+	Links    selfLinks `json:"links"`
+	ID       uint64    `json:"id"`
+	Name     string    `json:"name,omitempty"`
+	Provider string    `json:"provider,omitempty"`
+	Scheme   string    `json:"scheme,omitempty"`
+}
 
-	res := newUserResponse(srcID, u.Name).WithPermissions(u.Permissions)
-	if _, hasRoles := h.hasRoles(ctx, ts); hasRoles {
-		res.WithRoles(srcID, u.Roles)
+func newUserResponse(u *chronograf.User) *userResponse {
+	return &userResponse{
+		ID:       u.ID,
+		Name:     u.Name,
+		Provider: u.Provider,
+		Scheme:   u.Scheme,
+		Links: selfLinks{
+			Self: fmt.Sprintf("/chronograf/v1/users/%v", u.ID),
+		},
 	}
-	encodeJSON(w, http.StatusOK, res, h.Logger)
 }
 
-// RemoveSourceUser removes the user from the InfluxDB source
-func (h *Service) RemoveSourceUser(w http.ResponseWriter, r *http.Request) {
+func (s *Service) UserID(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	uid := httprouter.GetParamFromContext(ctx, "uid")
 
-	_, store, err := h.sourceUsersStore(ctx, w, r)
+	id := httprouter.GetParamFromContext(ctx, "id")
+	user, err := s.UsersStore.Get(ctx, id)
 	if err != nil {
+		Error(w, http.StatusBadRequest, err.Error(), s.Logger)
 		return
 	}
 
-	if err := store.Delete(ctx, &chronograf.User{Name: uid}); err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
-	}
-
-	w.WriteHeader(http.StatusNoContent)
+	res := newUserResponse(user)
+	encodeJSON(w, http.StatusOK, res, s.Logger)
 }
 
-// UpdateSourceUser changes the password or permissions of a source user
-func (h *Service) UpdateSourceUser(w http.ResponseWriter, r *http.Request) {
+func (s *Service) NewUser(w http.ResponseWriter, r *http.Request) {
```

Comment:
Make sure that there is consistency for all Service functions.  You changed this to s *Service; make sure all fields have the same change.

Reply:
we'll change it back to `h`, but what does `h` mean?

Reply:
I'd change it to s...

h is vestigial when I used to have the struct named Handlers